eng/New-TestResources.ps1

```diff
@@ -108,7 +129,10 @@ if ($ProvisionerApplicationId) {
     Log "Logging into service principal '$ProvisionerApplicationId'"
     $provisionerSecret = ConvertTo-SecureString -String $ProvisionerApplicationSecret -AsPlainText -Force
     $provisionerCredential = [System.Management.Automation.PSCredential]::new($ProvisionerApplicationId, $provisionerSecret)
-    $provisionerAccount = Connect-AzAccount -Tenant $TenantId -Credential $provisionerCredential -ServicePrincipal
+
+    Retry {

```

Comment:
Should we use Retry for all calls out to the network?

Reply:
I considered it and, no, because if some of the resource (e.g. resource group) was created trying to do it again would always fail, so there'd be no point.